src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs

```diff
@@ -156,7 +156,21 @@ private bool IsLambdaExpression(SemanticModel semanticModel, int position, Synta
             // open the builder if any overload takes a delegate at our argument position
             var inferredTypeInfo = typeInferrer.GetTypeInferenceInfo(semanticModel, position, cancellationToken: cancellationToken);
 
-            return inferredTypeInfo.Any(type => GetDelegateType(type, semanticModel.Compilation).IsDelegateType());
+            if (inferredTypeInfo.Any(type => GetDelegateType(type, semanticModel.Compilation).IsDelegateType()))
+            {
+                // In the following situation, the type inferrer will infer Task to support target type preselection
+                // Action a = Task.$$
+                // We need to explicitly exclude invocation/member access from suggestion mode
+                var previousToken = token.GetPreviousTokenIfTouchingWord(position);
```

Comment:
I'd much rather have the syntax check up front.  Asking if we're in a locatoin where you could have an inferred lambda.